x-pack/plugins/fleet/public/hooks/use_request/epm.ts

```diff
@@ -54,13 +54,16 @@ export function useGetReplacementCustomIntegrationsQuery() {
 }
 
 export function useGetCategoriesQuery(query: GetCategoriesRequest['query'] = {}) {
-  return useQuery<GetCategoriesResponse, RequestError>(['categories', query], () =>
-    sendRequestForRq<GetCategoriesResponse>({
-      path: epmRouteService.getCategoriesPath(),
-      method: 'get',
-      query,
-      version: API_VERSIONS.public.v1,
-    })
+  return useQuery<GetCategoriesResponse, RequestError>(
+    ['categories', query],
+    () =>
+      sendRequestForRq<GetCategoriesResponse>({
+        path: epmRouteService.getCategoriesPath(),
+        method: 'get',
+        query,
+        version: API_VERSIONS.public.v1,
+      }),
+    { retry: false, refetchOnWindowFocus: false }
```

Comment:
Instead of `retry: false` should we use a function to not retry on certain error? having retry is kind of the nice feature of `react-query` it could be nice to use it if we can

Reply:
Good call, I've refactored this to only retry 502's, as that's what we return for `PackageRegistryConnectionError` on the backend.